packages/app/obojobo-document-engine/src/scripts/oboeditor/components/node/editor-component.js

```diff
@@ -73,9 +86,59 @@ class Node extends React.Component {
 		this.props.editor.toggleEditable(false)
 	}
 
-	onClose() {
-		// Give cursor focus back to the editor
-		this.props.editor.toggleEditable(true)
+	// This method allows the keyboard shortcuts to open and close
+	// menus (insert and more info) to play nice with mouse users
+	onBlur(menu) {
+		// clear the open attribute on the top and bottom nodes
+		const nodes = Array.from(
+			Editor.nodes(this.props.editor, {
+				at: this.props.editor.selection || this.props.editor.prevSelection,
+				match: n => Element.isElement(n) && !n.subtype,
+				mode: 'lowest'
+			})
+		)
+
+		// Handle cases where no nodes are selected because the page is changing
+		if (nodes.length === 0) {
+			return this.props.editor.toggleEditable(true)
+		}
+
+		// If the first or last node's open attribute matches the menu we are
+		// blurring, we need to clear the open attribute
+		if(menu === nodes[0][0].open) {
```

Comment:
Do you mind explaining this a bit, just trying to wrap my head around it. Am I right that each slate node in the editor has an `open` attribute and that dictates if the More Info Box is open? What does `[0][0]` refer to?

Reply:
Yes, each node has an open attribute that can be either 'top', 'bottom', 'info', or null.  If it's not null, then the corresponding menu is open (top insert, bottom insert, or more info)
The Editor.nodes function returns a generator of NodeEntries, which is basically an array of [node, path] for each node it finds.  Therefore, nodes[0] is the first NodeEntry in the list, and nodes[0][0] is the actual slate node (nodes[0][1] is the path to the first returned node, which is used in the Transform)